tooling/tracer/src/lib.rs

```diff
@@ -15,10 +15,35 @@ use runtime_tracing::{Line, Tracer};
 
 use nargo::NargoError;
 
+/// A location in the source code: filename and line number (1-indexed).
+#[derive(PartialEq)]
+struct SourceLocation {
+    filepath: PathString,
+    line_number: isize,
+}
+
+impl SourceLocation {
+    /// Creates a source location that represents an unknown place in the source code.
+    fn create_unknown() -> SourceLocation {
+        SourceLocation { filepath: PathString::from_path(PathBuf::from("?")), line_number: -1 }
+    }
+}
+
+/// The result from step_debugger: the debugger either paused at a new location, reached the end of
+/// execution, or hit some kind of an error. Takes the error type as a parameter.
+enum DebugStepResult<Error> {
```

Comment:
Not sure if this is the best place for this code to live. But I do not have better suggestion